client/src/components/Settings-DisplayNames.vue

```diff
@@ -12,19 +18,21 @@ export default {
   props: ['value'],
   data () {
     return {
-      names: [],
-      edit: false
+      names: []
     }
   },
   methods: {
     updatePlayers () {
       this.names = []
+
       for (let i = 1; i <= this.value; i++) {
-        this.names.push('Player ' + i)
+        // replace name with random name function
+        this.names.push({ name: 'Player ' + i, editing: false })
       }
     },
-    updateName () {
-      console.log('updating')
+    updatingName (index) {
```

Comment:
What is the reason behind this. Is it because Vue won't update since you're changing an inner element of an array? Therefore the reference to the array is still the same?